src/misc/lv_async.c

```diff
@@ -65,6 +65,38 @@ lv_res_t lv_async_call(lv_async_cb_t async_xcb, void * user_data)
     return LV_RES_OK;
 }
 
+lv_res_t lv_async_call_cancel(lv_async_cb_t async_xcb, void * user_data)
+{
+    lv_timer_t * timer = lv_timer_get_next(NULL);
+    lv_res_t res = LV_RES_INV;
+
+    while(timer != NULL) {
+        /*Find async timer callback*/
+        if(timer->timer_cb == lv_async_timer_cb) {
+            lv_async_info_t * info = (lv_async_info_t *)timer->user_data;
+
+            /*Match user function callback and user data*/
+            if(info->cb == async_xcb && info->user_data == user_data) {
+                /*Record the timer node to be deleted*/
+                lv_timer_t * timer_del = timer;
+
+                /*Find the next timer node*/
+                timer = lv_timer_get_next(timer);
+
+                lv_timer_del(timer_del);
+                lv_mem_free(info);
+                res = LV_RES_OK;
+                continue;
```

Comment:
Why do we need this `continue` here? 

I see a problem here (maybe this `continue` is meant to solve it): if a timer is deleted the `timer` variable becomes invalid, so we can not use `lv_timer_get_next(timer);` to get the next timer from it. 

Usually I solve it by getting the next timer in the first line of the loop (where it's still surely valid) and just do `timer = timer_next` at the end of the loop.

Reply:
Done